[CI] Failure in PercolateQueryBuilderTests testToQuery

This test failed in the following build https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.6+oracle-periodic/668/console

The error is:

```
07:34:06 ERROR   0.40s J2 | PercolateQueryBuilderTests.testToQuery <<< FAILURES!
07:34:06    > Throwable #1: MapperParsingException[failed to parse]; nested: IllegalArgumentException[object field starting or ending with a [.] makes object resolution ambiguous: [ ]];
07:34:06    > 	at __randomizedtesting.SeedInfo.seed([7E7ABCF12BBA9DE:F01CA9F163386C34]:0)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.wrapInMapperParsingException(DocumentParser.java:176)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.parseDocument(DocumentParser.java:69)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentMapper.parse(DocumentMapper.java:277)
07:34:06    > 	at org.elasticsearch.percolator.PercolateQueryBuilder.doToQuery(PercolateQueryBuilder.java:410)
07:34:06    > 	at org.elasticsearch.index.query.AbstractQueryBuilder.toQuery(AbstractQueryBuilder.java:96)
07:34:06    > 	at org.elasticsearch.test.AbstractQueryTestCase.testToQuery(AbstractQueryTestCase.java:571)
07:34:06    > 	at java.lang.Thread.run(Thread.java:748)
07:34:06    > Caused by: java.lang.IllegalArgumentException: object field starting or ending with a [.] makes object resolution ambiguous: [ ]
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.splitAndValidatePath(DocumentParser.java:183)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.getMapper(DocumentParser.java:938)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.parseValue(DocumentParser.java:589)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.innerParseObject(DocumentParser.java:396)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.parseObjectOrNested(DocumentParser.java:373)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.parseObjectOrField(DocumentParser.java:465)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.parseObject(DocumentParser.java:504)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.innerParseObject(DocumentParser.java:383)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.parseObjectOrNested(DocumentParser.java:373)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.internalParseDocument(DocumentParser.java:93)
07:34:06    > 	at org.elasticsearch.index.mapper.DocumentParser.parseDocument(DocumentParser.java:66)
07:34:06    > 	... 41 more
```

The REPRO command is:

```
gradle :modules:percolator:test \
  -Dtests.seed=7E7ABCF12BBA9DE \
  -Dtests.class=org.elasticsearch.percolator.PercolateQueryBuilderTests \
  -Dtests.method="testToQuery" \
  -Dtests.security.manager=true \
  -Dtests.locale=es-MX \
  -Dtests.timezone=America/Costa_Rica
```

This reproduces locally for me.

--- a/test/framework/src/main/java/org/elasticsearch/ingest/RandomDocumentPicks.java
+++ b/test/framework/src/main/java/org/elasticsearch/ingest/RandomDocumentPicks.java
@@ -23,6 +23,8 @@ import com.carrotsearch.randomizedtesting.generators.RandomNumbers;
 import com.carrotsearch.randomizedtesting.generators.RandomPicks;
 import com.carrotsearch.randomizedtesting.generators.RandomStrings;
 
+import org.elasticsearch.common.Strings;
+
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
@@ -59,7 +61,7 @@ public final class RandomDocumentPicks {
         String fieldName;
         do {
             fieldName = randomString(random);
-        } while (fieldName.contains("."));
+        } while (fieldName.contains(".") || Strings.hasText(fieldName) == false);
         return fieldName;
     }
 

